pkg/apis/uiplugin/v1alpha1/types.go

```diff
@@ -31,20 +32,50 @@ type UIPluginList struct {
 	Items           []UIPlugin `json:"items"`
 }
 
-// +kubebuilder:validation:Enum=Dashboards
+// +kubebuilder:validation:Enum=Dashboards;Logging
 type UIPluginType string
 
 const (
-	// Dashboards deploys the Dashboards Dynamic Plugin for OpenShift Console.
+	// TypeDashboards deploys the Dashboards Dynamic Plugin for OpenShift Console.
 	TypeDashboards UIPluginType = "Dashboards"
+
+	// TypeLogging deploys the Logging View Plugin for OpenShift Console.
+	TypeLogging UIPluginType = "Logging"
 )
 
+// LoggingConfig contains options for configuring the logging console plugin.
+type LoggingConfig struct {
+	// LokiStack points to the LokiStack instance of which logs should be displayed.
+	// It always references a LokiStack in the "openshift-logging" namespace.
+	//
+	// +kubebuilder:validation:Required
+	LokiStack corev1.LocalObjectReference `json:"lokiStack"`
+
+	// LogsLimit is the max number of entries returned for a query.
+	//
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OCP Console Log Limit",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:ocpConsoleLogLimit"}
+	LogsLimit int `json:"logsLimit,omitempty"`
+
+	// Timeout is the max duration before a query timeout
+	//
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OCP Console Query Timeout",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:ocpConsoleTimeout"}
+	// +kubebuilder:validation:Pattern:="^([0-9]+)([smhd]{0,1})$"
+	Timeout string `json:"timeout,omitempty"`
+}
```

Comment:
The spec is missing the following two fields according to the [docs](https://github.com/openshift/logging-view-plugin?tab=readme-ov-file#configuration-values):
- `alertingRuleTenantLabelKey`
- `alertingRuleNamespaceLabelKey`

@jgbernalp are they omitted by design?

Reply:
They should not be omitted, they are seldom used. But they are still part of the allowed configuration.

Reply:
Currently this PR only includes the options that were present on the CLO API. I can add the other options as well, but then it might make sense to group them below another struct again. I'll take a stab at this tomorrow.